spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -141,5 +174,90 @@ module.exports = function (provider, setup) {
     });
   });
 
-  // TODO: add tests for tcpsocket.secure, accepting multiple.
+  it("Secures a socket", function (done) {
+    socket.connect('www.google.com', 443, function () {
+      var prepared;
+      // prepareSecure is Chrome specific, see freedom-for-chrome
+      socket.prepareSecure(function () {
+        prepared = true;
+      });
+      // TODO: timeout approach sucks, preferred approach:
```

Comment:
A bit bizarrely this conversation doesn't seem to have any reflection in
github... ah well...

Anyway, I'd suggest we add more details on prepareSecure to our API docs
(at:
https://github.com/freedomjs/freedom/blob/master/interface/core.tcpsocket.json)
so that we don't trip over this in the future. :)

On Tue, Mar 24, 2015 at 1:51 PM, soycode <notifications@github.com> wrote:

> In spec/providers/coreIntegration/tcpsocket.integration.src.js
> <https://github.com/freedomjs/freedom/pull/241#discussion_r27054444>:
>
> > @@ -141,5 +174,90 @@ module.exports = function (provider, setup) {
> >      });
> >    });
> >
> > -  // TODO: add tests for tcpsocket.secure, accepting multiple.
> > +  it("Secures a socket", function (done) {
> > +    socket.connect('www.google.com', 443, function () {
> > +      var prepared;
> > +      // prepareSecure is Chrome specific, see freedom-for-chrome
> > +      socket.prepareSecure(function () {
> > +        prepared = true;
> > +      });
> > +      // TODO: timeout approach sucks, preferred approach:
>
> Makes sense - for now I'll forget about prepareSecure then. Looks like the
> Chrome team is nearing fixing this on their end too, there's recent
> activity on the issue (
> https://code.google.com/p/chromium/issues/detail?id=467677).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom/pull/241/files#r27054444>.
>



-- 
Lucas Dixon | Google Ideas